native_client/swift/stt_ios_test/SpeechRecognitionImpl.swift

```diff
@@ -8,149 +8,72 @@
-        var destinationFormat = sourceFormat
-        destinationFormat.mSampleRate = 16000.0
-
-        var audioConverterRef: AudioConverterRef?
-        let createConverterStatus = AudioConverterNew(&sourceFormat, &destinationFormat, &audioConverterRef)
+    public func startMicrophoneRecognition() {
+        audioData = Data()
+        stream = try! model.createStream()
 
-        if (createConverterStatus != noErr) {
-            print("Error creating converter")
+        audioInput = AudioInput() { shorts in
+            self.bufferQueue.append(shorts)
         }
 
-        var quality = kAudioConverterQuality_Max
-
-        AudioConverterSetProperty(audioConverterRef!, kAudioConverterSampleRateConverterQuality, UInt32(MemoryLayout<UInt32>.size), &quality)
-
-        let blockBuffer = CMSampleBufferGetDataBuffer(sampleBuffer)
-
-        var pcmLength: Int = 0
-        var pcmData: UnsafeMutablePointer<Int8>?
-        let status: OSStatus = CMBlockBufferGetDataPointer(blockBuffer!, atOffset: 0, lengthAtOffsetOut: nil, totalLengthOut: &pcmLength, dataPointerOut: &pcmData)
-
-        if status != noErr {
-            print("Error getting something")
-        } else {
-            var input = FillComplexInputParm(source: pcmData!, sourceSize: UInt32(pcmLength))
-
-            let outputBuffer = malloc(pcmLength)
-            memset(outputBuffer, 0, pcmLength);
-
-            var outputBufferList = AudioBufferList()
-            outputBufferList.mNumberBuffers = 1
-            outputBufferList.mBuffers.mData = outputBuffer
-            outputBufferList.mBuffers.mDataByteSize = UInt32(Double(pcmLength) * destinationFormat.mSampleRate / sourceFormat.mSampleRate)
-            outputBufferList.mBuffers.mNumberChannels = 1
-
-            func inputDataProc(
-                inAudioConverter: AudioConverterRef,
-                ioNumberDataPacket: UnsafeMutablePointer<UInt32>,
-                ioData: UnsafeMutablePointer<AudioBufferList>,
-                outDataPacketDescription: UnsafeMutablePointer<UnsafeMutablePointer<AudioStreamPacketDescription>?>?,
-                inUserData: UnsafeMutableRawPointer?
-            ) -> OSStatus {
-                var inputPtr = inUserData!.load(as: FillComplexInputParm.self)
-
-                if (inputPtr.sourceSize <= 0) {
-                    ioNumberDataPacket.pointee = 1
-                    return -1
+        print("Started listening...")
+        audioInput!.start()
+
+        timer = Timer.scheduledTimer(
+            withTimeInterval: modelFeedInterval,
+            repeats: true
+        ) { _ in
+            if (!self.bufferQueue.isEmpty) {
+                let shorts = self.bufferQueue.removeFirst()
+                self.stream!.feedAudioContent(buffer: shorts)
+
+                // (optional) get partial result
+                let partialResult = self.stream!.intermediateDecode()
+                print(partialResult)
```

Comment:
I guess you can simply move this to a separate timer callback to `intermediateDecode` with a different cadence?

Reply:
true no problem